webapp/frontPage/js/modal.js

```diff
@@ -395,14 +402,26 @@ define(['jquery', 'magnific', 'util'], function($, magnific, util) {
             'X-LABKEY-CSRF': LABKEY.CSRF
           }
         }).success(function() {
-            window.location = LABKEY.ActionURL.buildURL("cds", "app.view?"); // set password should log user in automatically
-        }).error(function(e) {
-          var errorMsg = 'Create account failed. ';
-          if (e && e.responseJSON && e.responseJSON.errors && e.responseJSON.errors.length > 0) {
-            errorMsg = errorMsg + e.responseJSON.errors[0].message;
-          }
-          $('.create-account-modal .notifications p').html(errorMsg);
-        });
+
+            //sets NeedSurvey to true
+            $.ajax({
+              url: LABKEY.ActionURL.buildURL("cds", "updateNeedSurvey.api"),
+              method: 'POST',
+              data: {
+                'X-LABKEY-CSRF': LABKEY.CSRF
+              }
+            }).success(function() {
+              window.location = LABKEY.ActionURL.buildURL("cds", "app.view", null, {survey:true}); // set password should log user in automatically
+            }).error(function(e) {
+              var errorMsg = 'Unable to set NeedSurvey property to true';
+              createAccountError(e, errorMsg);
+              window.location = LABKEY.ActionURL.buildURL("cds", "app.view");
```

Comment:
maybe put redirect in a setTimeout call so user can briefly see the error (or maybe skip error at all since user won't make any sense of the msg)

Reply:
Good point! i've added error to console.log instead for dev troubleshooting.

Reply:
Good call. Another (one last) minor change: use console.error instead of console.log.

Reply:
Done!